### PR TITLE
Port Fix #5015, 5016 and #5028 to 2.0 branch + bugfixes #5163 #5146 #5175

### LINKED ISSUE #5015: Can't create filter for XREF/MREFs with integer ID in data item selection of data explorer
#### Reproduce
1. Upload this file:
https://drive.google.com/open?id=0B8iNw4kI1MA2VHRhcl84cU1Ec0E
2. Select entity 'cohorts' in data explorer
3. Click on 'imaging data' in data item selection
4. Type 'Brain' in popup

#### Expected
A list of items filtered by the word 'Brain'

#### Observed
Error! Illegal data type [INT] for operator [LIKE]

### LINKED ISSUE #5028: When setting lookupAttribute as false, it is still present in the dropdown menu
#### Reproduce
1. Upload this file: https://drive.google.com/open?id=0B8iNw4kI1MA2SlVybjc4M05sck0
2. Go to entity 'cohorts' in data explorer
3. Click on '+' button
4. Click on the field 'Imaging data' 

#### Expected
ID should not be present in the dropdown menu.

#### Observed
ID is still present in the dropdown menu.

### LINKED ISSUE #5146: Unique ints in forms won't pass validation
#### Reproduce
1. Upload *molgenis-app/src/test/resources/emx/emx_all_datatypes.xlsx*
2. Go to TypeTest entity
3. Add a new entity
4. Fill in required fields

#### Expected
Form can be submitted.

#### Observed
The *xint_unique* field won't validate, even if you enter a number. The validation message is: **Please enter a value**. The error message is: **Attribute [xint_unique] query value is of type [Double] instead of [Integer]**

### LINKED ISSUE #5163: Importing an EMX with an unknown refEntity is allowed
#### Reproduce
Import [emx_all_datatypes.xlsx](https://github.com/molgenis/molgenis/files/413270/emx_all_datatypes.xlsx)

#### Expected
Imports

#### Observed
``
java.lang.NullPointerException: null
	at org.molgenis.data.postgresql.PostgreSqlQueryGenerator.getSqlAttribute(PostgreSqlQueryGenerator.java:400) ~[molgenis-data-postgresql-2.0.0-SNAPSHOT.jar:na]
	at org.molgenis.data.postgresql.PostgreSqlQueryGenerator.lambda$getSqlCreateTable$2(PostgreSqlQueryGenerator.java:101) ~[molgenis-data-postgresql-2.0.0-SNAPSHOT.jar:na]
``

### LINKED ISSUE #5175: When opening the IndexManager plugin, Nullpointer happens
#### Reproduce
Open the Index Manager

#### Expected
I see All the entities I can index

#### Observed
Nullpointer

### PR DESCRIPTION
**Pull request by Mark-de-Haan**
### Bugfixes
- Fix #5163 Importing an EMX with an unknown refEntity is allowed
- Fix #5146 Unique ints in forms won't pass validation
- Fix #5175 When opening the IndexManager plugin, Nullpointer happens

### CODE REVIEW

File: molgenis-data-import/src/main/java/org/molgenis/data/importer/EmxMetaDataParser.java [hunk 1]
```diff
@@ -813,10 +814,14 @@ else if (attrName.startsWith(DESCRIPTION))
 				attr.setEnumOptions(enumOptions);
 			}
 
-			if (isReferenceType(attr) && StringUtils.isEmpty(emxRefEntity))
+			if (!emxDataType.equals("file"))
```
Comment (fdlk) [inline molgenis-data-import/src/main/java/org/molgenis/data/importer/EmxMetaDataParser.java]: shouldn't this be case insensitive equalsIgnoreCase()?

**Comment by fdlk** [comment]
I run into conflicts with changes made in RestController v1 for #5164

**Comment by tommydeboer** [comment]
I had no problem merging this with 2.0

**Comment by tommydeboer** [comment]
#5015 and #5028 aren't fixed yet in this PR.

**Pull request merged**